File: 프로그래머스/lv1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.js
function solution(numbers) {
    const sums = [];
    const answer = [];
    for(let i = 0; i < numbers.length; i++){
        for(let j = 0; j < numbers.length; j++){
           if(i !== j){
            sums.push(numbers[i] + numbers[j]); 
           }
        }
    }
    
    sums.forEach((e) => {
        if(!answer.includes(e)){
            answer.push(e);
        }
    })
    
    answer.sort((a,b) => a-b);
    return answer;
}

File: 프로그래머스/lv1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    public ArrayList<Integer> solution(int[] numbers) {
        Set<Integer> set = new HashSet<Integer>();

        for(int i = 0; i < numbers.length-1; i++){
            for(int j = i+1; j < numbers.length; j++){
                set.add(numbers[i] + numbers[j]);
            }
        }

        ArrayList<Integer> list = new ArrayList<>(set);
        Collections.sort(list);
        
        return list;
    }
}

/*
더하기 + : 교환법칙(순열 X)
서로 다른 인덱스 : 중복 X

*/